### PR TITLE
Ignore intern roles for traitor kill objective

### LINKED ISSUE #32271: remove intern roles from syndie kill objectives
it's very sad to watch new players trying to learn the game ask why they just got blown up in deadchat, intern roles are locked after so long anyways so I don't see why this can't be a thing

### PR DESCRIPTION
**Pull request by lzk228**
image

## About the PR

Intern, tech assist and research assist now can't be traitor kill objectives.

## Why / Balance

to resolve #32271 

## Technical details

add new list of jobs in PickRandomPersonComponent
then for pick check if any of humans has the job in list
remove human from list

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- tweak: Medical intern, technical assistant and research assistant now can't be traitor kill objectives.

### CODE REVIEW

**Comment by beck-thompson** [comment]
I always skip the objectives if its an intern as well. Really good idea!

**Comment by Cojoke-dot** [comment]
Service worker needs to be added if that pr to make them an intern job ever gets merged...

**Comment by Boaz1111** [comment]
intern should probably be clarified into medical intern in the CL, also, why does this not include cadets? Because they’re sec?

**Comment by lzk228** [comment]
> Because they’re sec?

you have to play 15 hours overall to play as cadet
also for me sounds weird that syndicate wouldn't want to kill security guys

**Comment by pheenty** [comment]
Add heads to this as well, heads should always fall under kill (not maroon) objective

**Comment by cohanna** [comment]
> Add heads to this as well, heads should always fall under kill (not maroon) objective

I don't see why?

**Comment by pheenty** [comment]
> > Add heads to this as well, heads should always fall under kill (not maroon) objective
> 
> I don't see why?

Because a couple of months ago antag difficulty has been correctly capped, but still one traitor can get "kill the RD and escape unrestrained" and the other can get something absurd like "kill or maroon the captain, kill or maroon the hos and escape unrestrained", because maroon objectives are considered easier (a maroon objective, say, a bartender, can die on their own and nobody would care, and in general killing a bartender is of course easier then a captain)

**Comment by Sarahon** [comment]
I don't really see it making much of a difference? If the person teaching the intern is the target, then the same issue persists? It's not like it's impossible to learn just because you're a possible target. The game overall just isn't really that beginner friendly and par for the course honestly. 

Like you can make all intern roles non-targeted, but the person teaching them may be targeted, so teaching halts once again.

File: Content.Server/Objectives/Components/PickRandomPersonComponent.cs [hunk 1]
```diff
@@ -8,4 +10,9 @@ namespace Content.Server.Objectives.Components;
 [RegisterComponent, Access(typeof(KillPersonConditionSystem))]
 public sealed partial class PickRandomPersonComponent : Component
 {
+    /// <summary>
+    /// List of jobs, that won't be kill objectives
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (deltanedas) [inline Content.Server/Objectives/Components/PickRandomPersonComponent.cs:16]: just [DataField]

File: Content.Server/Objectives/Systems/KillPersonConditionSystem.cs [hunk 2]
```diff
@@ -55,6 +56,12 @@ private void OnPersonAssigned(EntityUid uid, PickRandomPersonComponent comp, ref
 
         // no other humans to kill
         var allHumans = _mind.GetAliveHumansExcept(args.MindId);
+
+        // filter jobs that we don't want to be kill objectives
+        allHumans = allHumans
+            .Where(human => !_job.MindTryGetJobId(human, out var jobId) || !comp.IgnoredJobs.Contains(jobId))
+            .ToList();
```
Comment (deltanedas) [inline Content.Server/Objectives/Systems/KillPersonConditionSystem.cs:63]: use RemoveAll instead of linq funny

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by Everturning** [comment]
> At that point why not just make most, if not all service roles non-target since they're also usually where new players are recommended?

this is a false dichotomy, as even if new players are recommended to do service, that does not mean all service players are new players just because thats recommended.

**Comment by Sarahon** [comment]
> > At that point why not just make most, if not all service roles non-target since they're also usually where new players are recommended?
> 
> this is a false dichotomy, as even if new players are recommended to do service, that does not mean all service players are new players just because thats recommended.

and not all interns are new either? It's way more likely a new player is gonna be service vs a department intern. Like it's globally recommended people start janitor for their first few times playing I thought?

**Comment by slarticodefast** [comment]
Hey, we had a maintainer vote and the result was to close this PR for the following reasons:
- Being an intern does not equal being a new player. You could have lots of playtime in other departments or on other servers.
- Being killed by antagonists is part of the game, the story it is telling and not always something negative. It can even be a good thing for newer player to learn that early.
- There are other antagonists where such a safety feature does not exist (nukies, revs, eventually Wizard)

Thank you for your contribution.

**Comment by lzk228** [comment]
weh, okay, it was good experience tho

**Pull request closed without merging**